src/compat/db_json.hpp

```diff
@@ -55,12 +61,19 @@ enum DB_JSON_TYPE
   DB_JSON_BOOL,
 };
 
+enum class JSON_PATH_TYPE
```

Comment:
I think that "enum JSON_PATH_TYPE" without "class" suffices

Reply:
I think with C++11 it's better to use enum class because: complete type safety, well-defined underlying type, scope issues, and forward declaration